src/components/TilePropertiesMenu/TilePropertiesMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import { useStore } from '../../store/store';
+import { useState } from 'react';
+import './TilePropertiesMenu.scss';
+import { createPortal } from 'react-dom';
+import { useClickOutside } from '../../hooks';
+import { Vector3 } from 'three';
+
+export const TilePropertiesMenu = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [locationTitle, setLocationTitle] = useState('');
+
+  const selected = useStore((state) => state.selected);
+  const modalRef = useClickOutside(() => setModalOpen(false));
+  const setCamera = useStore((state) => state.setCamera);
+
+  const locations = useStore((state) => state.locations);
+
+  const saveLocation = useStore((state) => state.saveLocation);
+
+  const handleAdd = (t: string, x: number, y: number) => {
+    saveLocation(t, x, y);
+    selected!.object.userData.update();
+    setModalOpen(false);
+    setLocationTitle('');
+  };
+
+  return (
+    <>
+      <button
+        className="TilePropertiesMenu__button"
+        disabled={!selected}
+        onClick={() => {
+          setModalOpen(true);
+        }}
+      >
+        📍
+      </button>
+
+      {modalOpen &&
+        createPortal(
+          <div className="TilePropertiesMenu__modal" ref={modalRef}>
+            <input
+              type="text"
+              value={locationTitle}
+              onChange={(e) => setLocationTitle(e.target.value)}
+              placeholder="Enter location title"
+              className="TilePropertiesMenu__input"
+            />
+            <button
+              className="TilePropertiesMenu__modal-confirm"
+              disabled={!selected || !locationTitle.trim()}
+              onClick={() => {
+                const { x, y } = selected!.object.position;
+                handleAdd(locationTitle, x, y);
+              }}
+            >
+              💾
+            </button>
+
+            <h3>Saved Locations</h3>
+
+            <ul>
+              {Object.entries(locations).map(([title, { positionX, positionY }]) => (
+                <li key={title} className="TilePropertiesMenu__locations-item">
+                  {title}: ({positionX.toFixed(2)}, {positionY.toFixed(2)})
+                  <button
+                    onClick={() => {
+                      const newPosition = new Vector3(positionX, positionY, 10);
+                      setCamera(newPosition as Vector3, new Vector3(positionX, positionY, 0) as Vector3);
```

Comment:
this doesn't work because it only saves the desired position to the store. But it does not actually change the position of the camera. For that you need to retrieve the camera from three.js (using the useThree() hook) and then change the position of that accordingly.

You can look at the useLayoutEffect() hook in the Map.tsx file to see how it can be done. In there, the camera's position is changed to cameraHistory.position with the code `camera.position.copy(cameraHistory.position);`. However, you will get an error when trying to do the same here, because the TilePropertiesMenu component is not inside the three.js context. Maybe you can find a workaround for that? If not, just leave it like this for now and we can work on that later